src/middleware/dataset.ts

```diff
@@ -0,0 +1,14 @@
+import { Next } from 'koa';
+// eslint-disable-next-line import/no-cycle
+import { AppContext, AppMiddleware } from '../app';
```

Comment:
Relates to https://github.com/benmosher/eslint-plugin-import/issues/1453. But I have an idea on a refactor to get around it.